### ISSUE TITLE
input.onChange function is inconsistently named on wrapped Field component

### ISSUE THREAD

**Issue by jamesmoss** [erikras/redux-form#3377]
Currently there are are several functions that get passed into a component wrapped by `Field` via the `input` prop (see http://redux-form.com/7.0.3/docs/api/Field.md/#input-props):

- `input.onBlur`
- `input.onChange`
- `input.onDragStart`
- `input.onDrop`
- `input.onFocus`

Because these are all prefixed with `on` you might think that these are for callbacks that fire within your component's lifecycle, but they are actually for effecting changes to state. e.g `props.input.onChange('Hello world')` will set the value of the field to `Hello world`. This is pretty confusing, especially as in other places in Redux Form this doesn't happen. e.g the `change` prop passed into your component wrapped by `reduxForm`.

This is compounded by the fact that are props you can pass into the `Field` component which _are_ callbacks (and which don't get passed down into your wrapped component).

I propose dropping the `on` prefix for all these functions:

- `input.blur`
- `input.change`
- `input.dragStart`
- `input.drop`
- `input.focus`

This would make the API a bit more consistent and definitely less confusing.

**Comment by gustavohenke** [comment]
I agree that it would be a nice change to the API, but it's a breaking one 😐

**Comment by erikras** [comment]
The reason for the design choice as it stands now is so that you can do:
```jsx
const renderInput = ({ input, placeholder, meta: { error } }) =>
  <div>
    <input {...input} placeholder={placeholder}/>
//          ^^^^^^^^
    {error && <span>{error}</span>}
  </div>
```
That feels really elegant to me. Consider how much more ugly that would look with the API proposed here.

You do have a point about confusion with callbacks (although they _are_ callbacks, just not the ones you're referring to).

**Comment by spinningarrow** [comment]
This is pretty confusing. When I call the prop as `input.onBlur(event)` from my form field (I'm using [React Native Elements' FormInput][forminput]) it just crashes with this error:

```
ExceptionsManager.js:65 One of the sources for assign has an enumerable key on the prototype chain. Are you trying to assign a prototype property? We don't allow it, as this is an edge case that we do not support. This error is a performance optimization and not spec compliant.
```

But if I call `input.onBlur()` without passing in the `event` arg, everything is fine. I can't figure out why. It also conflicts with the flow type checks for `onBlur` which expect it to be called with an argument. https://github.com/erikras/redux-form/blob/f036f8773eb9fa5cd743b829bcfc752dfa64d25c/src/FieldProps.types.js.flow#L31

[forminput]: https://react-native-training.github.io/react-native-elements/API/forms/
